src/main/java/gyro/google/storage/BucketResource.java

```diff
@@ -0,0 +1,625 @@
+            Bucket bucket = new Bucket();
+
+            if (changedFieldNames.contains("acl")) {
+                bucket.setAcl(getAcl() == null ? null : getAcl().stream().map(BucketAccessControlConfiguration::toBucketAccessControl).collect(Collectors.toList()));
+            }
+
+            if (changedFieldNames.contains("labels")) {
+                bucket.setLabels(getLabels());
+            }
+
+            if (changedFieldNames.contains("location")) {
+                bucket.setLocation(getLocation());
+            }
+
+            if (changedFieldNames.contains("defaultEventBasedHold")) {
+                bucket.setDefaultEventBasedHold(getDefaultEventBasedHold());
+            }
+
+            if (changedFieldNames.contains("cors")) {
+                bucket.setCors(getCors().stream().map(BucketCors::toBucketCors).collect(Collectors.toList()));
+            }
+
+            if (changedFieldNames.contains("billing")) {
+                bucket.setBilling(getBilling() == null ? null : getBilling().toBucketBilling());
+            }
+
+            if (changedFieldNames.contains("encryption")) {
+                bucket.setEncryption(getEncryption() == null ? null : getEncryption().toBucketEncryption());
+            }
+
+            if (changedFieldNames.contains("iamConfiguration")) {
+                bucket.setIamConfiguration(getIamConfiguration() == null ? null : getIamConfiguration().toBucketIamConfiguration());
+            }
+
+            if (changedFieldNames.contains("lifecycle")) {
+                bucket.setLifecycle(getLifecycle() == null ? null : getLifecycle().toLifecycle());
+            }
+
+            if (changedFieldNames.contains("logging")) {
+                bucket.setLogging(getLogging() == null ? null : getLogging().toBucketLogging());
+            }
+
+            if (changedFieldNames.contains("retentionPolicy")) {
+                bucket.setRetentionPolicy(getRetentionPolicy() == null ? null : getRetentionPolicy().toBucketRententionPolicy());
+            }
+
+            if (changedFieldNames.contains("storageClass")) {
+                bucket.setStorageClass(getStorageClass());
+            }
+
+            if (changedFieldNames.contains("versioning")) {
+                bucket.setVersioning(getVersioning() == null ? null : getVersioning().toBucketVersioning());
+            }
+
+            if (changedFieldNames.contains("website")) {
+                bucket.setWebsite(getWebsite() != null ? getWebsite().toBucketWebsite() : null);
+            }
+
+            copyFrom(storage.buckets().patch(getName(), bucket).execute());
+        } catch (GoogleJsonResponseException e) {
+            throw new GyroException(e.getDetails().getMessage());
+        }
+    }
+
+    @Override
+    public void delete(GyroUI ui, State state) throws Exception {
+        try {
+            Storage storage = createClient(Storage.class);
+            storage.buckets().delete(getName()).execute();
+        } catch (IOException e) {
```

Comment:
This should handle the GoogleJsonResponseException so as to get better details if it errors
```suggestion
        } catch (GoogleJsonResponseException e) {
```

Reply:
Done.